src/holder.rs

```diff
@@ -1,7 +1,79 @@
 use crate::sync::atomic::AtomicPtr;
 use crate::{Domain, HazPtrObject, HazPtrRecord};
+use std::mem::{ManuallyDrop, MaybeUninit};
 use std::sync::atomic::Ordering;
 
+pub struct HazardPointerArray<'domain, F, const N: usize> {
+    haz_ptrs: [ManuallyDrop<HazardPointer<'domain, F>>; N],
+}
+
+impl<'domain, F, const N: usize> HazardPointerArray<'domain, F, N> {
+    pub fn hazard_pointers<'array>(&'array mut self) -> [&'array mut HazardPointer<'domain, F>; N] {
+        let mut out: [MaybeUninit<&'array mut HazardPointer<'domain, F>>; N] =
+            [(); N].map(|_| MaybeUninit::uninit());
```

Comment:
Ugh, we really want [`array::each_mut`](https://doc.rust-lang.org/std/primitive.array.html#method.each_mut) here, but it's unstable. Would let us get rid of the `MaybeUninit` and have the body of this be just:
```rust
self.haz_ptrs.each_mut().map(|v| &mut **v)
```